### ISSUE TITLE
Button and Buttongroup accessibility improvements

### ISSUE THREAD

**Issue by PetarMetodiev** [telerik/kendo-angular#1859]
## I'm submitting a...

* Suggestion for improvement  

## Current behavior

 * Button: when `[togglable]=true` the button does not receive a role
 * Buttongroup: when a `button` in `buttongroup` does **not** have `[togglable]=true` input it receives `role="checkbox"` and `aria-checked` attribute 

## Expected behavior

 * Button: when `[togglable]=true` the button should receive `role="checkbox"`
 * Buttongroup: when a `button` in `buttongroup` does **not** have `[togglable]=true` input it **should not** receive `role="checkbox"` and `aria-checked` attribute

## Minimal reproduction of the problem with instructions

https://stackblitz.com/edit/angular-4ee19z
## Environment

Package versions:

```
latest version
```

Browser:

- Chrome (desktop) version XX
- Chrome (Android) version XX
- Chrome (iOS) version XX
- Firefox version XX
- Safari (desktop) version XX
- Safari (iOS) version XX
- IE version XX
- Edge version XX

## Additional information
 * Button: in order to tell a screen reader that the button is checked, it should have attribute `aria-checked`, which according to the [spec](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/checkbox/checkbox-1.html) should also come with `role="checkbox"`
 * Buttongroup: currently the buttongroup has only `single` and `multiple` selection modes, but can be initialised with non-togglable buttons. As a result the non-togglable buttons will receive `role="checkbox"` and `aria-checked` attribute which does not make sense. A suggestion for resolving the issue - a third selection mode could be added, e.g. `[selection]="'none'"`. Additionally there should be a check if the buttons in the buttongroup have `[togglable]="true"` and according to the selection mode to throw exception or silently override their settings.

cc @tsvetomir @valchev

**Comment by tsvetomir** [comment]
I agree on the expected behavior - individual buttons should get a `checkbox` role only when they're toggleable, regardless of their placement.

The button group should apply single and multiple modes only to the toggleable buttons, if any. 
Nothing should prevent users using plain buttons or a mix in a group, and a mandatory "none" selection mode would be a burden for them.

**Comment by PetarMetodiev** [comment]
The buttons now work as advised by @tsvetomir : 
A toggleable button always receives `role="checkbox"` and `aria-checked` attributes regardless of their placement.
A buttongroup does not add `role="checkbox"` and `aria-checked` attributes to non-toggleable buttons inside it.

**Comment by ipeshev** [comment]
Fix available in [latest development version](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/).

**Comment by tsvetomir** [comment]
Fixed in v4.3.0
